packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -0,0 +1,1121 @@
+  ///
+  /// When set to true, a border with [borderWidth], [borderRadius] and the
+  /// appropriate border color will render. Otherwise, no border will be
+  /// rendered.
+  final bool renderBorder;
+
+  /// The border color to display when the toggle button is enabled and not
+  /// selected.
+  ///
+  /// If this property is set to null, then
+  /// ToggleButtonTheme.of(context).borderColor is used. If
+  /// [ToggleButtonsThemeData.borderColor] is also null, then
+  /// Theme.of(context).colorScheme.onSurface is used.
+  final Color borderColor;
+
+  /// The border color to display when the toggle button is selected.
+  ///
+  /// If this property is set to null, then
+  /// ToggleButtonTheme.of(context).selectedBorderColor is used. If
+  /// [ToggleButtonsThemeData.selectedBorderColor] is also null, then
+  /// Theme.of(context).colorScheme.primary is used.
+  final Color selectedBorderColor;
+
+  /// The border color to display when the toggle button is disabled.
+  ///
+  /// If this property is set to null, then
+  /// ToggleButtonTheme.of(context).disabledBorderColor is used. If
+  /// [ToggleButtonsThemeData.disabledBorderColor] is also null, then
+  /// Theme.of(context).disabledBorderColor is used.
+  final Color disabledBorderColor;
+
+  /// The width of the border surrounding each toggle button.
+  ///
+  /// This applies to both the greater surrounding border, as well as the
+  /// borders dividing each toggle button.
+  ///
+  /// To render a hairline border (one physical pixel), set borderWidth to 0.0.
+  /// See [BorderSide.width] for more details on hairline borders.
+  ///
+  /// To omit the border entirely, set [renderBorder] to false.
+  ///
+  /// If this property is set to null, then
+  /// ToggleButtonTheme.of(context).borderWidth is used. If
+  /// [ToggleButtonsThemeData.borderWidth] is also null, then
+  /// a width of 1.0 is used.
+  final double borderWidth;
+
+  /// The radii of the border's corners.
+  ///
+  /// If this property is set to null, then
+  /// ToggleButtonTheme.of(context).borderRadius is used. If
+  /// [ToggleButtonsThemeData.borderRadius] is also null, then
+  /// the buttons default to non-rounded borders.
+  final BorderRadius borderRadius;
+
+  bool _isFirstIndex(int index, int length, TextDirection textDirection) {
+    return index == 0 && textDirection == TextDirection.ltr
+        || index == length - 1 && textDirection == TextDirection.rtl;
+  }
+
+  BorderRadius _getEdgeBorderRadius(int index, int length, TextDirection textDirection, ToggleButtonsThemeData toggleButtonsTheme) {
+    final BorderRadius resultingBorderRadius = borderRadius ?? toggleButtonsTheme.borderRadius ?? BorderRadius.zero;
+
+    if (_isFirstIndex(index, length, textDirection)) {
+      return BorderRadius.only(
+        topLeft: resultingBorderRadius.topLeft,
+        bottomLeft: resultingBorderRadius.bottomLeft,
+      );
+    } else if (
+      index == length - 1 && textDirection == TextDirection.ltr ||
```

Comment:
These expressions might be a tiny bit clearer if parentheses were added:
```
(index == length - 1 && textDirection == TextDirection.ltr ) || 
(index == 0 && textDirection == TextDirection.rtl)
```

Reply:
I realized I could refactor these out to a `_isLastIndex` method and did so instead